src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1303 @@
+            final Broadcast<ReadMetadata> broadcastMetadata,
+            final List<Interval> intervals,
+            final JavaRDD<GATKRead> reads ) {
+        final Broadcast<List<Interval>> broadcastIntervals = ctx.broadcast(intervals);
+        final List<QNameAndInterval> qNameAndIntervalList =
+                reads
+                        .mapPartitions(readItr ->
+                                new MapPartitioner<>(readItr,
+                                        new QNameFinder(broadcastMetadata.value(),
+                                                broadcastIntervals.value())), false)
+                        .collect();
+        broadcastIntervals.destroy();
+
+        final HopscotchUniqueMultiMap<String, Integer, QNameAndInterval> qNamesMultiMap =
+                new HopscotchUniqueMultiMap<>(params.assemblyToMappedSizeRatioGuess*qNameAndIntervalList.size());
+        qNamesMultiMap.addAll(qNameAndIntervalList);
+        return qNamesMultiMap;
+    }
+
+    /** write template names and interval IDs to a file. */
+    private static void dumpQNames( final String qNameFile,
+                                    final PipelineOptions pipelineOptions,
+                                    final Collection<QNameAndInterval> qNames ) {
+        try (final OutputStreamWriter writer = new OutputStreamWriter(new BufferedOutputStream(
+                BucketUtils.createFile(qNameFile, pipelineOptions)))) {
+            for (final QNameAndInterval qnameAndInterval : qNames) {
+                writer.write(qnameAndInterval.toString() + "\n");
+            }
+        } catch (final IOException ioe) {
+            throw new GATKException("Can't write qname intervals file " + qNameFile, ioe);
+        }
+    }
+
+    /**
+     * Identify funky reads that support a hypothesis of a breakpoint in the vicinity, group the reads,
+     * and declare a breakpoint interval where there is sufficient density of evidence.
+     */
+    @VisibleForTesting static List<Interval> getIntervals(
+            final Params params,
+            final Broadcast<ReadMetadata> broadcastMetadata,
+            final SAMFileHeader header,
+            final JavaRDD<GATKRead> reads,
+            final Locations locations ) {
+        // find all breakpoint evidence, then filter for pile-ups
+        final int maxFragmentSize = broadcastMetadata.value().getMaxMedianFragmentSize();
+        final List<SAMSequenceRecord> contigs = header.getSequenceDictionary().getSequences();
+        final int nContigs = contigs.size();
+        final int minMapQ = params.minEvidenceMapQ;
+        final int minMatchLen = params.minEvidenceMatchLength;
+        final int minEvidenceCount = params.minEvidenceCount;
+        final JavaRDD<BreakpointEvidence> evidenceRDD =
+                reads
+                    .filter(read ->
+                            read.getMappingQuality() >= minMapQ &&
+                            read.getCigar().getCigarElements()
```

Comment:
It's implemented as
`return samRecord.getCigar() == null ? Collections.emptyList() : samRecord.getCigar().getCigarElements();`
so it won't be any faster today, but it *could* be faster with revised code and I don't see how it could ever be slower, so I'll make that change.